test/config/test/browser-configs.test.ts

```diff
@@ -303,3 +306,143 @@ test('can enable browser-cli options for multi-project workspace', async () => {
   expect(projects[1].config.browser.enabled).toBe(true)
   expect(projects[1].config.browser.headless).toBe(true)
 })
+
+function getCliConfig(options: UserConfig, cli: string[], fs: Record<string, string> = {}) {
+  const root = resolve(process.cwd(), `vitest-test-${crypto.randomUUID()}`)
+  useFS(root, {
+    ...fs,
+    'basic.test.ts': /* ts */`
+      import { test } from 'vitest'
+      test('basic', () => {
+        expect(1).toBe(1)
+      })
+    `,
+    'vitest.config.ts': /* ts */ `
+      export default {
+        test: {
+          reporters: [
+            {
+              onInit(vitest) {
+                const browser = vitest.config.browser
+                console.log(JSON.stringify({
+                  browser: {
+                    headless: browser.headless,
+                    browser: browser.enabled,
+                    ui: browser.ui,
+                  },
+                  workspace: vitest.projects.map(p => {
+                    return {
+                      name: p.name,
+                      headless: p.config.browser.headless,
+                      browser: p.config.browser.enabled,
+                      ui: p.config.browser.ui,
+                    }
+                  })
+                }))
+                // throw an error to avoid running tests
+                throw new Error('stop')
+              },
+            },
+          ],
+          ...${JSON.stringify(options)}
+        }
+      }
+    `,
+  })
+  return runVitestCli(
+    {
+      nodeOptions: {
+        env: {
+          CI: 'false',
+          GITHUB_ACTIONS: undefined,
+        },
+      },
+    },
+    '--root',
+    root,
+    '--no-watch',
+    ...cli,
+  )
+}
+
+test('[e2e] CLI options correctly override inline workspace options', async () => {
+  const vitest = await getCliConfig({
+    workspace: [
+      {
+        test: {
+          name: 'unit',
+        },
+      },
+      {
+        test: {
+          name: 'browser',
+          browser: {
+            enabled: true,
+            headless: true,
+            instances: [
+              {
+                browser: 'chromium',
+              },
+            ],
+          },
```

Comment:
This config should actually throw an error that `preview` provider doesn't support `headless: true`. 🤔 

Using `instances: [{ browser: 'chromium' }]`  makes this look like `playwright` was intended but it actually resolves to `preview`.

Reply:
The preview-headless error is thrown by the provider when tests start to run, this test bails out before that